maestro-ios-driver/src/main/kotlin/xcuitest/XCTestDriverClient.kt

```diff
@@ -170,6 +170,44 @@ class XCTestDriverClient(
         installer.close()
     }
 
+    fun shake(){
+        logger.trace("Executing shake gesture on iOS simulator")
+
+        try {
+            // Create ProcessBuilder for executing AppleScript
+            val processBuilder = ProcessBuilder(
+                "osascript",
+                "-e", """
+                tell application "Simulator"
+                    activate
+                    delay 0.5 -- Give simulator time to come to foreground
+                    tell application "System Events"
+                        tell process "Simulator"
+                            -- Access the Hardware menu and click Shake Gesture
+                            click menu item "Shake" of menu "Device" of menu bar 1
+                        end tell
+                    end tell
+                end tell
+            """.trimIndent()
+            )
+
+            // Execute the command
+            val process = processBuilder.start()
+            val exitCode = process.waitFor()
+
+            if (exitCode != 0) {
+                val errorStream = process.errorStream.bufferedReader().readText()
+                logger.error("Failed to execute shake gesture. Exit code: $exitCode, Error: $errorStream")
+                throw IOException("Failed to execute shake gesture on simulator. Exit code: $exitCode")
```

Comment:
The catch block below will re-throw this with this message wrapped in the same message again:
"Failed to execute shake gesture on simulator: Failed to execute shake gesture on simulator. Exit code: $exitCode"

Reply:
Thanks for the review. https://github.com/mobile-dev-inc/Maestro/pull/2259/commits/68212e09855366bfc31d91ae4c3e56b2c9cafd48